plugins/lookup/passwordstore.py

```diff
@@ -156,7 +159,7 @@ def parse_params(self, term):
         # I went with the "traditional" param followed with space separated KV pairs.
         # Waiting for final implementation of lookup parameter parsing.
         # See: https://github.com/ansible/ansible/issues/12255
-        params = term.split()
+        params = shlex.split(term)
```

Comment:
Instead of changing how the string parsing works, we should start using the proper lookup plugin argument parsing. That has been implemented some years ago already.

Reply:
I'd love to put that in - do you happen to have a link / example where this is used?

Reply:
This is one plugin which uses it: https://github.com/ansible-collections/community.general/blob/main/plugins/lookup/tss.py

Line 116 is the most important part - it must be called at the beginning of the `run()` method, so that all options are processed. After that, `self.get_option()` can be used (like in https://github.com/ansible-collections/community.general/blob/main/plugins/lookup/tss.py#L120).

The most complicated part is doing this in a way that's backwards compatible. One plugin doing that is hashi_vault, see for example https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/lookup/hashi_vault.py#L258 - see here for it is done there: https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/lookup/hashi_vault.py#L595-L597

CC @briantist - I'm not sure anymore whether you implemented this, or whether it was already present before. In case you implemented it, maybe you can weight in on potential pitfalls.

In any case, this should be done in its own PR, since it is a more complex undertaking.

For this PR (YAML support), it's probably best to remove the `shlex` part, and call `pass` manually to insert the YAML key in the tests instead of using the `passwordstore` plugin.

Reply:
`hashi_vault` traditionally used term string k=v pairs only. I updated that in https://github.com/ansible-collections/community.general/pull/23 to support "real" params, and did it in a backward-compatible way, to fully use config manager as @felixfontein suggested. I highly recommend going that route. It also gives you native support for `env:` and `ini:` entries to get params from those sources in a consistent way without reimplementing that code.

I also agree it should be in its own PR. It seems to me that replacing `str.split()' with `shlex.split()` is a breaking change, as it would behave differently; either way that's easier to evaluate in its own PR.

Reply:
Should I open an issue for migrating to the better term parsing style?